### PR TITLE
Replacement for calc module with SymPy

### LINKED ISSUE #278: Replacement for 'calc' module
The calc module depends on DuckDuckGo instant answers (which is due to be reimplemented in the search module in https://github.com/goavki/phenny/issues/271) and Wolfram Alpha (which now requires users to pay for API requests), in the two commands implemented. Both commands no longer work as DDG does not handle calculator requests correctly, while WA requires payment. Using [SymPy](http://www.sympy.org), write a replacement for the 'calc' module.

### PR DESCRIPTION
**Pull request by gstark0**
Fixes #278 
Features:
- Support for operators such as +, -, *, /, ** (power, or ^)
- Logarithms
- Trigonometry, Polynomials
- Trigonometric, hyperbolic, exponential, roots, logarithms, absolute value, spherical harmonics, factorials and gamma functions, zeta functions, polynomials, special functions etc (more here - http://www.sympy.org/en/features.html)
- Arbitrary precision integers, rationals, and floats
- Finding unknown variables

Now, you can also use expressions without multiplication sign so e.g '3x' instead of '3 * x' or '2(x-1)' instead of '2 * (x-1)' etc.

Some simple examples of use:
.c 3+2
.c 2+x=4
.c 2^x=4
.c sin(90)
etc

### CODE REVIEW

**Comment by wei2912** [comment]
Your code should remove the "wa" command and provide a few more tests. Could you schedule a session with the mentors to test the bot (since SymPy works rather differently from the previous code)? Thanks!

**Comment by wei2912** [comment]
This is to ensure that it provides the functionality we need.

**Comment by wei2912** [comment]
Also, please add SymPy to requirements.txt, so that the build will pass.

**Comment by gstark0** [comment]
screen shot 2017-12-31 at 16 52 16
Here are some examples on real channel :)

**Comment by wei2912** [comment]
Could you put the change to git.py in another PR? You should not bundle unrelater changes together into one commit.

**Comment by wei2912** [comment]
Also, can you explain the regex?

File: modules/git.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def do_POST_unsafe(self, data):
                         '{:}',
                         commit['url'][:commit['url'].rfind('/') + 7]
                     )
-                    msgs.append(non_trunc.format(truncate(non_trunc.format(''), commit['message']))
+                    msgs.append(non_trunc.format(truncate(non_trunc.format(''), commit['message'])))
```
Comment (Androbin) [inline modules/git.py]: This one's my fault. Fixing in #297.

**Comment by gstark0** [comment]
@wei2912 
Regex basically does the same thing I could do in next 10 lines of code, it just matches different cases and groups it so it can determine if it's the correct expression, like 2x 2( 2xy etc.

Full explanation you can find here -> https://stackoverflow.com/a/35930060

**Pull request merged**